### ISSUE TITLE
On the Jetson orin nx to detect train model the map/map50 is zero??  why?

### ISSUE THREAD

**Issue by mathsky** [ultralytics/ultralytics#8374]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Jetson orin nx 16G  cuda11.4 torch 2.0.0 torchvision 0.15.0

 Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size
        2/3       2.1G      1.271      2.818     0.9879        463        640: 100%|██████████| 54/54 [00:18<00:00,  2.99it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 4/4 [00:00<00:00,  6.14it/s]
                   all         61       1652          0          0          0          0

**Comment by glenn-jocher** [comment]
@mathsky hello! It looks like your model isn't detecting any objects, which is why your precision (P), recall (R), mAP50, and mAP50-95 are all zero. This could be due to several reasons, such as:

- **Incorrect labels:** Ensure your dataset is correctly labeled and that the labels are in a format that YOLOv8 can understand.
- **Learning rate:** Sometimes the learning rate might be too high or too low. Try adjusting it.
- **Model architecture:** Make sure the model architecture is suitable for your dataset's complexity.
- **Insufficient training:** If you've only trained for a few epochs, consider training for more epochs.

Here's a quick check you can perform:

```bash
# Validate using the last checkpoint to see if the issue persists
yolo val model=runs/train/exp/weights/last.pt data=coco128.yaml
```

If the problem continues, please provide more details about your training setup, and we can investigate further. Keep up the great work with your Jetson Orin NX setup! 👍

**Comment by glenn-jocher** [comment]
@mathsky hi there! It's intriguing that your dataset works on a GTX 1050 but not on the Jetson Orin NX. 🤔 Since the mAP is zero after 300 epochs, it might be an issue specific to the Jetson environment. Here are a couple of things to check:

1. **Data Paths**: Verify that the dataset paths in `VOC.yaml` are correct and accessible on the Jetson.
2. **CUDA Compatibility**: Ensure that the CUDA version is compatible with the PyTorch version you're using.
3. **Hardware Utilization**: Monitor the GPU usage during training to see if it's being utilized properly.

You could also try a quick validation with:

```bash
yolo val model=runs/train/exp/weights/last.pt data=VOC.yaml
```

If the issue persists, please share the training logs, and we'll take a closer look! Keep us posted. 😊

**Comment by mathsky** [comment]
@glenn-jocher hi !  I use the sanm config and datasets to Val  the  problem is as follows：

**Comment by glenn-jocher** [comment]
@mathsky hi there! 👋 It seems like there's an issue with your validation results showing zeros across the board. This could be due to a variety of reasons, such as incorrect dataset paths or issues with the model weights. Could you please double-check your dataset paths in your configuration file to ensure they're correct and accessible? Also, trying a different set of weights, such as the pretrained ones, might help identify if the issue lies with the training process. 

Here's a quick command to validate with pretrained weights, just to check everything else is set up correctly:

```bash
yolo val model=yolov8n.pt data=your_dataset.yaml
```

Let's start from there and see how it goes! 🚀

**Comment by glenn-jocher** [comment]
@mathsky Hi! 👋 Great to see your validation results looking good! It seems like your setup is correct if you're getting valid metrics. If you encounter any further issues or have questions, feel free to reach out. Happy detecting with YOLOv8! 🚀

**Comment by glenn-jocher** [comment]
Hi @mathsky! 🌟 It looks like your training on the Jetson Orin NX might not be proceeding as expected if you're seeing zeros across your metrics. A common reason for this could be related to the learning rate or the dataset not being properly loaded or utilized during training. 

Could you try adjusting the learning rate and double-checking your dataset paths in your YAML file? Also, ensure your dataset is correctly labeled and accessible. Here's a quick snippet to check dataset loading:

```python
from ultralytics import YOLO, utils

# Load your dataset
dataset = YOLO('datasets/myvoc.yaml')

# Check if it's loaded correctly
print(utils.dataset_info(dataset))
```

Let's start with these steps and see if there's any improvement. Keep pushing forward! 💪

**Comment by mathsky** [comment]
Hi @glenn-jocher! I used this code to check dataset loading:
from ultralytics import YOLO, utils

# Load your dataset
dataset = YOLO('datasets/myvoc.yaml')

# Check if it's loaded correctly
print(utils.dataset_info(dataset))


There is some error 
![1709083512127](https://github.com/ultralytics/ultralytics/assets/9279539/ce01fdfa-baf2-4f3d-852d-a927cc57c2ef)

**Comment by glenn-jocher** [comment]
@mathsky hi there! 😊 It looks like there might be an issue with how the dataset is being loaded or with the path specified in your `myvoc.yaml`. Could you double-check the paths in your YAML file to ensure they're correct and point to the right directories? Also, make sure that the `YAML` file structure follows the expected format:

```yaml
# Example of dataset YAML structure
train: /path/to/train/images
val: /path/to/val/images

# Classes
nc: 20  # number of classes
names: ['class1', 'class2', ..., 'class20']
```

If the paths are correct and you're still facing issues, it might be helpful to share the content of your `myvoc.yaml` for a closer look. Keep up the great work! 🚀

**Comment by mathsky** [comment]
Hi ! @glenn-jocher  I find the function check_det_dataset() ,dataset is right !  I try to use the CPU mode train the data the error look here :
![1709181424285](https://github.com/ultralytics/ultralytics/assets/9279539/bbbad9b3-66d2-49c8-98e4-e1f815ccfb8c)

**Comment by mathsky** [comment]
Hi @glenn-jocher  Thanks for your help , I plan to train on a PC, but the training speed on Jetsong is not fast enough，Thanks!

**Comment by glenn-jocher** [comment]
@mathsky hi there! 👋 You're welcome! Training on a PC with a more powerful GPU can indeed speed up the process significantly compared to the Jetson. If you encounter any issues or have questions about setting up your training environment on PC, feel free to reach out. Happy training! 🚀

**Comment by alesbani** [comment]
Hello! I have exactly the same issue on Orin AGX with the VOC dataset. I would like to add that the problem occurs only with yolov8n, while with the small model yolov8s the metrics turn out to be correct.

`

`

**Comment by glenn-jocher** [comment]
@alesbani hi there! 😊 It's interesting that you're only experiencing this issue with yolov8n and not with yolov8s on your Orin AGX. This might hint at a configuration or resource allocation difference between the two models. Have you tried adjusting the batch size or learning rate specifically for yolov8n? Lightly tweaking these parameters could potentially help. Also, ensuring that your VOC dataset is correctly formatted and accessible to the model during training is crucial. Here's a quick snippet to check your dataset loading:

```python
from ultralytics import YOLO

# Load your dataset
model = YOLO('path/to/yolov8n.yaml')  # Make sure paths inside yolov8n.yaml are correct

# Quick dataset check
print(model.dataset_info)
```

Keep us posted on your progress! 🌟

**Comment by alesbani** [comment]
update: solved using ultralytics/ultralytics:latest-jetson docker container instead of custom setup.

**Comment by glenn-jocher** [comment]
@alesbani that's fantastic news! 🎉 Glad to hear you found a solution by using the ultralytics/ultralytics:latest-jetson Docker container. It's a neat approach to ensure a consistent environment for running YOLOv8 models. If you have any more insights or questions down the line, feel free to share. Happy detecting!
